crates/sparrow-session/src/session.rs

```diff
@@ -42,6 +49,16 @@ pub struct ExecutionOptions {
     pub max_batch_size: Option<usize>,
     /// Whether to run execute as a materialization or not.
     pub materialize: bool,
+    // History or Snapshot results.
+    pub results: Results,
+    /// The changed since time. This is the minimum timestamp of changes to events.
+    /// For historic queries, this limits the output points.
+    /// For snapshot queries, this limits the set of entities that are considered changed.
+    pub changed_since_time_s: Option<i64>,
+    /// The final at time. This ist he maximum timestamp output.
+    /// For historic queries, this limits the output points.
+    /// For snapshot queries, this determines the time at which the snapshot is produced.
+    pub final_at_time_s: Option<i64>,
```

Comment:
Maybe: "The time to produce the snapshot at. This _is the_* maximum timestamp output." 

We don't allow `final_at_time` with `historic` queries. Right now, we have three entity behaviors: `ALL, FINAL, FINAL_AT_TIME`, and the last implies `snapshot` behavior.

Reply:
Ah. We should probably allow with history it as a way to limit the upper bound of times. This is documenting the internal options, so I'm OK with it describing how we intend it to work.